legacy/jacorb/src/main/java/org/jboss/as/jacorb/JacORBExtension.java

```diff
@@ -185,11 +185,83 @@ protected static void registerTransformers_1_3(final SubsystemRegistration subsy
+        builder.getAttributeBuilder().addRejectCheck(RejectAttributeChecker.DEFINED,
+                JacORBSubsystemDefinitions.PERSISTENT_SERVER_ID);
+        convertDefaultAttributeValues(builder);
         TransformationDescription.Tools.register(builder.build(), subsystem, ModelVersion.create(1, 3, 0));
     }
+
+    private static void convertDefaultAttributeValues(final ResourceTransformationDescriptionBuilder builder) {
+        builder.getAttributeBuilder().setValueConverter(new ChangeUndefinedDefaultValueConverter(new ModelNode().set("JBoss")),
+                JacORBSubsystemDefinitions.ORB_NAME);
+        builder.getAttributeBuilder().setValueConverter(new ChangeUndefinedDefaultValueConverter(new ModelNode().set(5)),
+                JacORBSubsystemDefinitions.ORB_CONN_RETRIES);
+        builder.getAttributeBuilder().setValueConverter(new ChangeUndefinedDefaultValueConverter(new ModelNode().set(500)),
+                JacORBSubsystemDefinitions.ORB_CONN_RETRY_INTERVAL);
+        builder.getAttributeBuilder().setValueConverter(new ChangeUndefinedDefaultValueConverter(new ModelNode().set(0)),
+                JacORBSubsystemDefinitions.ORB_CONN_CLIENT_TIMEOUT);
+        builder.getAttributeBuilder().setValueConverter(new ChangeUndefinedDefaultValueConverter(new ModelNode().set(0)),
+                JacORBSubsystemDefinitions.ORB_CONN_SERVER_TIMEOUT);
+        builder.getAttributeBuilder().setValueConverter(new ChangeUndefinedDefaultValueConverter(new ModelNode().set(2048)),
+                JacORBSubsystemDefinitions.ORB_CONN_OUTBUF_SIZE);
+        builder.getAttributeBuilder().setValueConverter(new ChangeUndefinedDefaultValueConverter(new ModelNode().set(-1)),
+                JacORBSubsystemDefinitions.ORB_CONN_OUTBUF_CACHE_TIMEOUT);
+        builder.getAttributeBuilder().setValueConverter(
+                new ChangeUndefinedDefaultValueConverter(new ModelNode().set(Integer.MAX_VALUE)),
+                JacORBSubsystemDefinitions.ORB_CONN_MAX_SERVER_CONNECTIONS);
+        builder.getAttributeBuilder().setValueConverter(new ChangeUndefinedDefaultValueConverter(new ModelNode().set(24)),
+                JacORBSubsystemDefinitions.ORB_CONN_MAX_MANAGED_BUF_SIZE);
+        builder.getAttributeBuilder().setValueConverter(new ChangeUndefinedDefaultValueConverter(new ModelNode().set(10)),
+                JacORBSubsystemDefinitions.POA_QUEUE_MIN);
+        builder.getAttributeBuilder().setValueConverter(new ChangeUndefinedDefaultValueConverter(new ModelNode().set(100)),
+                JacORBSubsystemDefinitions.POA_QUEUE_MAX);
+        builder.getAttributeBuilder().setValueConverter(new ChangeUndefinedDefaultValueConverter(new ModelNode().set(5)),
+                JacORBSubsystemDefinitions.POA_REQUEST_PROC_POOL_SIZE);
+        builder.getAttributeBuilder().setValueConverter(new ChangeUndefinedDefaultValueConverter(new ModelNode().set(32)),
+                JacORBSubsystemDefinitions.POA_REQUEST_PROC_MAX_THREADS);
```

Comment:
I apologize; I didn't do a cross-reference of the attributes where you changed the default value vs the ones JacORBSubsystemAdd is rejecting. If I had I would have realized what was going on and given different input.

These attributes are rejected on a current server if they are defined, but they still have their previous default values if they end up on a running server. Our attribute description metadata doesn't quite account for that concept, but I don't want to change the default value for this. I'd rather just have the text description of the attribute say that it will be rejected if configured on a running server.

This subsystem has 2 uses now -- legacy slave support on a DC, and a short term migration aid on current servers. Performing correctly for the legacy slave support case should be higher priority, as that's something users may use for many years. That means the default values should define what they truly are on the processes that support them, i.e. the legacy servers. The compatibility subsystem aspect should be to help make it easy to migrate, but it shouldn't redefine the attributes.